uvm/bootstrap/com/untangle/uvm/UvmContextSelector.java

```diff
@@ -0,0 +1,248 @@
+import org.apache.logging.log4j.core.layout.PatternLayout;
+import org.apache.logging.log4j.core.net.Facility;
+import org.apache.logging.log4j.core.net.Protocol;
+import org.apache.logging.log4j.core.selector.ContextSelector;
+import org.apache.logging.log4j.status.StatusLogger;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Custom ContextSelector implementation to create multiple logger context for apps
+ */
+public class UvmContextSelector implements ContextSelector {
+
+    private final ConcurrentHashMap<String, LoggerContext> contexts = new ConcurrentHashMap<>();
+    private static final StatusLogger LOGGER = StatusLogger.getLogger();
+    public static final String UVM_LOG = "uvm";
+    private static final UvmContextSelector INSTANCE;
+    private static final ThreadLocal<String> THREAD_LOG_INFO = new InheritableThreadLocal<>();
+
+    static {
+        INSTANCE = new UvmContextSelector();
+    }
+
+    /**
+     * instance() Provides the UvmContextSelector singleton
+     * @return UvmContextSelector
+     */
+    public static UvmContextSelector instance() {
+        return INSTANCE;
+    }
+
+    /**
+     * Overrides getContext method of ContextSelector interface
+     * @param fqcn fqcn
+     * @param loader loader
+     * @param currentContext boolean isCurrentContext
+     * @return LoggerContext
+     */
+    @Override
+    public LoggerContext getContext(String fqcn, ClassLoader loader, boolean currentContext) {
+        try{
+            String contextName = THREAD_LOG_INFO.get();
+            synchronized (contexts) {
+                if (!contexts.containsKey(contextName)) {
+                    UvmLoggerContext context = new UvmLoggerContext(contextName);
+                    context.start();
+                    contexts.put(contextName, context);
+                }
+            }
+            return contexts.get(contextName);
+        } catch (Exception e) {
+            LOGGER.error("Exception: ", e);
+        }
+        return contexts.get(UVM_LOG);
+    }
+
+    /**
+     * Overrides getContext method of ContextSelector interface
+     * @param fqcn fqcn
+     * @param loader loader
+     * @param currentContext boolean isCurrentContext
+     * @param configLocation configLocation
+     * @return LoggerContext
+     */
+    @Override
+    public LoggerContext getContext(String fqcn, ClassLoader loader, boolean currentContext, URI configLocation) {
+        return getContext(fqcn, loader, currentContext);
+    }
+
+    /**
+     * Overrides removeContext method of ContextSelector interface
+     * @param context LoggerContext
+     */
+    @Override
+    public void removeContext(LoggerContext context) {
+        synchronized (contexts) {
+            LoggerContext remove = contexts.remove(context.getName());
```

Comment:
No need to store the return value in `remove` variable, just do 
``` 
contexts.remove(context.getName());

Reply:
removed assignment of return variable